### PR TITLE
[bitnami/drupal] Add VIB tests

### PR DESCRIPTION
**Pull request by migruiz4**
### Description of the change

The main objective of this PR is to publish our Bitnami Drupal container using VMware Image Builder. In order to do that, several changes are included:

- Increasing the existing test coverage of the asset by adding Goss tests.
- Update verify and publish VIB pipeline's definitions.

### Benefits

- Ensuring higher quality of the container catalog.
- Increased pool of assets completely handled by VMware Image Builder.

### Possible drawbacks

Automated tests could introduce additional flakiness to the CI/CD.

### Applicable issues

NA

### CODE REVIEW

File: .vib/drupal/goss/vars.yaml [hunk 1]
```diff
@@ -0,0 +1,24 @@
...
+    - /opt/bitnami/drupal/sites/default/settings.php
+    - /opt/bitnami/drupal/sites/default/files/.htaccess
+directories:
+  - mode: "0775"
+    paths:
+      - /bitnami/drupal
+      - /opt/bitnami/drupal
+      - /opt/bitnami/drupal/sites/default/files
+      - /opt/bitnami/drupal/themes
+      - /opt/bitnami/drupal/modules
```
Comment (FraPazGal) [inline .vib/drupal/goss/vars.yaml:19]: We are not actually setting the permissions for these 2 dirs, I would move them to another element of the `.Vars.directories` array without the `mode` variable.

File: .vib/drupal/goss/drupal.yaml
Comment (FraPazGal) [inline .vib/drupal/goss/drupal.yaml:1]: We are adding the `libphp` component but not actively checking it. To do so, I suggest checking that Apache's config is loading the module, wdyt?

```
$ cat /opt/bitnami/apache2/conf/httpd.conf | grep libphp 
LoadModule php_module modules/libphp.so
```

File: .vib/drupal/goss/vars.yaml [hunk 2]
```diff
@@ -0,0 +1,24 @@
+binaries:
+  - apachectl
+  - gosu
+  - mysql
+  - php
+  - render-template
+  - drush
+files:
+  - paths:
+    - /opt/bitnami/drupal/sites/default/settings.php
```
Comment (FraPazGal) [inline .vib/drupal/goss/vars.yaml:10]: We should also check the custom Drupal config performed at the postunpack stage is properly set:

https://github.com/bitnami/containers/blob/91b3f028c95686ef8112f748d39c63fb20871164/bitnami/drupal/10/debian-11/rootfs/opt/bitnami/scripts/drupal/postunpack.sh#L50

Something similar happens with the changes to the following files:

- [PHP config file](https://github.com/bitnami/containers/blob/91b3f028c95686ef8112f748d39c63fb20871164/bitnami/drupal/10/debian-11/rootfs/opt/bitnami/scripts/drupal/postunpack.sh#L54) `/opt/bitnami/php/etc/php.ini`
- [Drupal .htaccess file](https://github.com/bitnami/containers/blob/91b3f028c95686ef8112f748d39c63fb20871164/bitnami/drupal/10/debian-11/rootfs/opt/bitnami/scripts/drupal/postunpack.sh#L127) `/opt/bitnami/apache/conf/vhosts/htaccess/drupal-htaccess.conf`

We need to do these kind of checks to make sure our added logic works as expected and no changes will break them (files not existing anymore, components not correctly added, etc)

**Review by FraPazGal — COMMENTED** [review]
Nice PR @migruiz4! I commented on a couple of things added here, please check them whenever you have some time.

File: .vib/drupal/goss/vars.yaml (hunk 1 above)
Comment (migruiz4) [reply .vib/drupal/goss/vars.yaml:21]: Thanks for the suggestion! I applied this change 👍

File: .vib/drupal/goss/drupal.yaml
Comment (migruiz4) [reply .vib/drupal/goss/drupal.yaml:1]: I added a test checking for the modules enabled/disabled at https://github.com/bitnami/containers/blob/main/bitnami/drupal/10/debian-11/rootfs/opt/bitnami/scripts/apache-modphp/postunpack.sh

File: .vib/drupal/goss/vars.yaml (hunk 2 above)
Comment (migruiz4) [reply .vib/drupal/goss/vars.yaml:10]: Added extra tests verifying the content of `php.ini`, `drupal-htaccess.conf`, `settings.php` and `httpd.conf`

File: .vib/drupal/goss/drupal.yaml [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+    exists: true
+    filetype: file
+    contains:
+      - "Include \"/opt/bitnami/apache/conf/bitnami/php.conf\""
+group:
+  daemon:
+    exists: true
+user:
+  daemon:
+    exists: true
```
Comment (joancafom) [inline .vib/drupal/goss/drupal.yaml:65]: This is related to Apache and not Drupal

File: .vib/drupal/goss/vars.yaml [hunk 4]
```diff
@@ -0,0 +1,31 @@
...
+    - /opt/bitnami/drupal/sites/default/files/.htaccess
```
Comment (joancafom) [inline .vib/drupal/goss/vars.yaml:11]: There is no longer need to have these files here:

- There is a more advanced tests for `/opt/bitnami/drupal/sites/default/settings.php` already in `drupal.yaml`
- You are checking `/opt/bitnami/drupal/sites/default/settings.php`, which should result from `/opt/bitnami/drupal/sites/default/files/.htaccess`. If the latter does not exist, the changes should neither be present in the former 😁

**Review by joancafom — COMMENTED** [review]
Nice first PR, I left you some comments as a review 🚀

File: .vib/common/goss/templates/check-apache-libphp.yaml [hunk 5]
```diff
@@ -0,0 +1,15 @@
+command:
+  # Check explicitly enabled and disabled modules
+  check-loaded-modules:
```
Comment (joancafom) [inline .vib/common/goss/templates/check-apache-libphp.yaml:3]: Rename it in order not to collide with other defined tests

```suggestion
  check-apache-libphp-loaded-modules:
```

File: .vib/drupal/goss/drupal.yaml [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+    filetype: file
+    contains:
+      - /^memory_limit = 256M/
+  /opt/bitnami/apache/conf/bitnami/php.conf:
+    exists: true
+    filetype: file
+    contains:
+      - "AddType application/x-httpd-php .php"
+      - "DirectoryIndex index.html index.htm index.php"
```
Comment (joancafom) [inline .vib/drupal/goss/drupal.yaml:43]: This belongs to `apache-modphp` and should be moved to the common test you've implemented above

**Review by joancafom — COMMENTED** [review]
Thanks for the changes! Please, remove the triggers from the code and I'll proceed to approve the PR 😁

File: .vib/drupal/goss/vars.yaml [hunk 7]
```diff
@@ -0,0 +1,21 @@
+binaries:
+  - apachectl
+  - gosu
```
Comment (joancafom) [inline .vib/drupal/goss/vars.yaml:3]: Remove gosu, as it is no longer part of the containers.

**Review by joancafom — APPROVED** [review]
LGTM! Thanks for addressing the changes! 🚀

**Pull request merged**